scripts/speech_recognition/convert_to_tarred_audio_dataset.py

```diff
@@ -686,14 +673,51 @@ def _create_shard(self, entries, target_dir, shard_id, manifest_folder):
             # Need the following replacement as long as WebDataset splits on first period
             base = base.replace('.', '_')
             squashed_filename = f'{base}{ext}'
-            if squashed_filename not in count:
-                self._write_to_tar(tar, audio_filepath, squashed_filename)
-                to_write = squashed_filename
-                count[squashed_filename] = 1
-            else:
-                to_write = base + "-sub" + str(count[squashed_filename]) + ext
+
+            if self.config.slice_with_offset:
+                if squashed_filename not in count:
+                    count[squashed_filename] = 1
+
+                entry_offset = str(entry['offset']).split('.')
+                if len(entry_offset) == 1:
+                    entry_offset.append('0')
+                elif len(entry_offset) > 2:
+                    raise ValueError(
+                        f"The offset for the entry with audio_filepath '{entry['audio_filepath']}' is incorrectly provided ({entry['offset']})."
```

Comment:
What is the correct format? A comment here with an example + clarifying message in the exception would be helpful.